include/picongpu/plugins/randomizedParticleMerger/VoronoiCell.hpp

```diff
@@ -0,0 +1,367 @@
+            if ( parentNumMacroParticles > minMacroParticlesToDivide )
+            {
+                expectedNumMacroParticles = numMacroParticles * ratioKeptParticles;
+            }
+            else
+            {
+                // TODO: this equation works, but there may be a better one (see the notebook)
+                float_X undividedCellCoeff = (
+                    parentExpectedNumMacroParticles + parentNumMacroParticles ) / 2.0_X;
+                float_X currentExpectedNumMacroParticles =
+                    numMacroParticles * undividedCellCoeff / parentNumMacroParticles;
+                expectedNumMacroParticles = currentExpectedNumMacroParticles;
+            }
+        }
+
+        /** determine in which of the two sub-Voronoi cells a particle falls */
+        HDINLINE
+        int32_t getSubVoronoiCell(
+            const floatD_X position,
+            const float3_X momentum
+        ) const
+        {
+            const float_X valParticle = splittingStage == VoronoiSplittingStage::position ?
+                position[splittingComponent] :
+                momentum[splittingComponent];
+            const float_X meanVoronoi = splittingStage == VoronoiSplittingStage::position ?
+                meanPositionValue[splittingComponent] :
+                meanMomentumValue[splittingComponent];
+            return
+                valParticle < meanVoronoi ?
+                lowerCellId :
+                higherCellId;
+        }
+
+        /** Counting parameters that are necessary before processing vornoi cell:
+         *  mean values and expected number of macro particles
+         *
+         * @param minMacroParticlesToDivide min number of macroparticles in a cell
+         *                                  such that the cell is always subdivided
+         * @param ratioKeptParticles ratio of particles that are kept on average
+         */
+
+        /** auxillary function for getting the mean squared deviation in position or momentum */
+        HDINLINE
+        float_X getMaxValueSpread2(
+            uint8_t& component,
+            const uint8_t dimension
+        ) const
+        {
+            const float3_X meanValue2 = splittingStage == VoronoiSplittingStage::position ?
+                  meanPositionValue * meanPositionValue :
+                  meanMomentumValue * meanMomentumValue;
+
+            const float3_X valueSpread2 = splittingStage == VoronoiSplittingStage::position ?
+                  meanPositionSquaredValue - meanValue2 :
+                  meanMomentumSquaredValue - meanValue2;
+
+            /* find component of most spread in position */
+            component = 0;
+            float_X maxValueSpread2 = valueSpread2[0];
+            for( uint8_t i = 1; i < dimension; i++ )
+            {
+                if( valueSpread2[i] > maxValueSpread2 )
+                {
+                    maxValueSpread2 = valueSpread2[i];
+                    component = i;
```

Comment:
`component` is never used.

So you can rewrite the `if` with `maxValueSpread2 = math::max( maxValueSpread2, valueSpread2[ i ] )`

Reply:
It is result of the function, by reference. But should be documented as that.

Reply:
Yes, I thought it is a function local variable.